Sources/MongoSwift/ConnectionString.swift

```diff
@@ -188,4 +192,26 @@ internal class ConnectionString {
 
         return hosts
     }
+
+    internal func setCredential(from credential: Credential) throws {
```

Comment:
nit: I'd probably just omit the argument label here since the one argument we pass in is the credential so I think the relationship is fairly clear at the call site. alternatively you could name the method something like `setAuthenticationOptions`